### PR TITLE
Initial plugin infrastructure and supervisor-in-iframe system

### LINKED ISSUE #627: CICD build race condition building wasm-loader ui files
There is a race condition intermittently impacting only Ubuntu 2004 builds on CICD on the `dev/plugins` branch and any branch off of that. I'm not able to reproduce this locally.

image

### PR DESCRIPTION
**Pull request by brandonfancher**
This is a long-running dev feature branch. Once we're satisfied with the code quality and functionality, and we know we're not breaking anything, we'll merge.

This branch will be locked down. PRs should be made against this branch and reviewed before being merged.

Remaining:
- [x] Fix #627
- [x] Clean up common resources
    - [x] Dedupe sources of truth for RPC and keyConversions. Combine with messaging code into `common-lib`
    - [x] Two directories under `common`: `resources` (not built) and `packages` (built)
    - [x] Move `thirdParty` into `resources` (#635)
- [x] Add WIT parsing (@Velua, #639)
- [x] Document the demo1 -> Supervisor -> loader -> demo2 path and dev-facing Supervisor API (@cool-ant, #599)
- [x] Remove broken UIs (#626)  [Tag for broken UIs](https://github.com/gofractally/psibase/tree/broken-UIs)
- [x] Clean up `CMakeLists.txt` (#634)
- [x] Determine whether we merge Demo Apps into `main` for now, and where and in what state they're eligible for merge if so (@cool-ant #653)
- [x] James address Steven's comments

### CODE REVIEW

**Comment by James-Mart** [comment]
Added @swatanabe specifically for a review on `CMakeLists.txt`

File: CMakeLists.txt [hunk 1]
```diff
@@ -544,6 +576,16 @@ psibase_package(
...
 )
 
+psibase_package(
+    OUTPUT ${SERVICE_DIR}/SupervisorSys.psi
+    NAME SupervisorSys
+    VERSION ${PSIBASE_VERSION}
+    PACKAGE_DEPENDS NftSys ProxySys AccountSys
+    SERVICE supervisor-sys
+        DATA ${CMAKE_CURRENT_SOURCE_DIR}/services/user/SupervisorSys/ui/dist /
+    DEPENDS SupervisorSys_js ${CMAKE_CURRENT_SOURCE_DIR}/services/user/SupervisorSys/ui/dist/index.html
```
Comment (swatanabe) [inline CMakeLists.txt:586]: `${SupervisorSys_js_dep}` would be better

File: CMakeLists.txt [hunk 2]
```diff
@@ -217,47 +217,80 @@ add_test(
...
+            --target wasm32-unknown-unknown
+            --manifest-path ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/Cargo.toml 
+            --target-dir    ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/target
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${COMPONENT_BIN_DIR}
+        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/target/wasm32-unknown-unknown/release/${OUTPUT_FILE} ${OUTPUT_FILEPATH}
+    )
+    # Create a custom target to ensure `${OUTPUT_FILEPATH}` is built before `${TARGET_NAME}`
+    add_custom_target(${TARGET_NAME} DEPENDS ${OUTPUT_FILEPATH})
+    # Expose `${TARGET_NAME}` and its dependency to the parent scope for further use
+    set(${TARGET_NAME}_DEP ${TARGET_NAME} PARENT_SCOPE)
```
Comment (swatanabe) [inline CMakeLists.txt:285]: When used in other custom commands, this adds only a target-level dependency. A file-level dependency is also required to ensure that out-of-dateness propagates correctly.

File: CMakeLists.txt [hunk 3]
```diff
@@ -217,47 +217,80 @@ add_test(
...
+    add_js_target(${ROW} index.html ${common-misc-resources} ${common-fonts} ${CommonSysCommonLib_js_DEP} ${CommonSysLoaderLib_js_DEP})
 endforeach()
 
+# A function that can be used to add a new target that builds a wasm component from rust
+set(COMPONENT_BIN_DIR ${CMAKE_CURRENT_BINARY_DIR}/components)
+function(add_rs_component TARGET_TUPLE OUTPUT_FILE)
+    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\1 PATH ${TARGET_TUPLE})
+    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\2 TARGET_NAME ${TARGET_TUPLE})
+    set(OUTPUT_FILEPATH ${COMPONENT_BIN_DIR}/${OUTPUT_FILE})
+    file(GLOB_RECURSE SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/src/*)
```
[the remark this answers is not in the document]
Comment (James-Mart) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 1 above)
Comment (James-Mart) [reply CMakeLists.txt]: fixed

File: CMakeLists.txt (hunk 2 above)
Comment (James-Mart) [reply CMakeLists.txt]: fixed

**Review by James-Mart — APPROVED** [review]

**Pull request merged**